dscache/convert_test.go

```diff
@@ -233,12 +236,152 @@ func TestConvertLogbookAndRefsMissingDsref(t *testing.T) {
 	}
 }
 
+// Test a logbook missing refs from dsref will add them when making the dscache
+// For now, this should only happen for users that were modifying their logbook or who created
+// datasets before logbook existed. In the future, it shouldn't be necessary to explicitly
+// handle this case. Eventually dsrefs will go away and only dscache will be used in this
+// manner.
+func TestConvertLogbookAndRefsMissingFromLogbook(t *testing.T) {
+	run := NewDscacheTestRunner()
+	defer run.Delete()
+
+	ctx := context.Background()
+
+	peerInfo := testPeers.GetTestPeerInfo(0)
+	book := makeFakeLogbook(ctx, t, "test_user", peerInfo.PrivKey)
+
+	// Dsrefs has a third reference that is not in logbook.
+	dsrefs := []repo.DatasetRef{
+		repo.DatasetRef{
```

Comment:
For future reference, the go compiler can infer the types of elements of a slice in declaration like this, you can use just curly braces.